### PR TITLE
Add Unsqueeze Module

### PR DESCRIPTION
**Pull request by pengsun**
This module, nn.Unsqueeze, is the "inverse" operation of nn.Squeeze. It inserts a singleton dim (i.e., dimension 1) at specified position.

### CODE REVIEW

File: Unsqueeze.lua [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+local function _checkPos(pos)
+   local pos = pos or error('the position to insert singleton dim not specified')
+   assert(type(pos) == 'number')
+   assert(pos > 0)
+   return pos
+end
+
+local function _assertTensor(t)
+   assert(t or torch.isTensor(t), "Unsqueeze only works on tensor")
```
Comment (fmassa) [inline Unsqueeze.lua]: this should be `t and torch.isTensor(t)` right ?

Comment (fmassa) [inline Unsqueeze.lua]: actually, only `torch.isTensor(t)` is enough.

**Comment by szagoruyko** [comment]
the name is ambiguous I think

**Comment by fmassa** [comment]
Another point, the forward pass could leverage [`nn.utils.addSingletonDimension`](https://github.com/torch/nn/blob/master/utils.lua#L133). This would avoid code redundancy.

File: Unsqueeze.lua (hunk 1 above)
Comment (pengsun) [inline Unsqueeze.lua]: @fmassa Yeah, you're right, thanks for pointing out this. Will modify the code.

**Comment by pengsun** [comment]
@szagoruyko Initially, I named it as `InsertSingletonDim`. Latter at torch 7 gitter chatting room @soumith mentioned the name `Unsqueeze`, which I think is not bad, since it recalls one its connection to `nn.Squeeze`. So what shall we call this module in the end? Maybe nn.AddSingletonDimension ?

**Comment by pengsun** [comment]
@fmassa Ah... didn't notice this function at all, since it is not mentioned at the doc... Yeah, I would try to wrap it.

**Comment by soumith** [comment]
Thanks @pengsun , nicely done.

**Comment by pengsun** [comment]
Happy to contribute :-)

**Comment by Atcold** [comment]
:+1:

**Pull request merged**